lib/rubocop/cop/lint/nested_double_quotes_in_interpolation.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for double-quoted strings that contain interpolations with
+      # double-quoted strings, which can be hard to read and understand.
+      #
+      # @example
+      #
+      #   # bad
+      #   "#{success? ? "yes" : "no"}"
+      #
+      #   # good
+      #   "#{success? ? 'yes' : 'no'}"
+      class NestedDoubleQuotesInInterpolation < Cop
+        MSG =
+          'Nesting double-quotes makes strings hard to read. ' \
+          'Either switch to single-quotes or disable this cop.'.freeze
```

Comment:
I'd certainly drop the part about disabling the cop and merge two sentences into one, using the imperative voice.